src/constraint_tree.jl

```diff
@@ -211,11 +212,26 @@ function variables(; keys::AbstractVector{Symbol}, bounds = nothing)
         length(bounds) == length(keys) ? bounds :
         error("lengths of bounds and keys differ for allocated variables")
     ConstraintTree(
-        k => Constraint(value = LinearValue(Int[i], Float64[1.0]), bound = b) for
-        ((i, k), b) in zip(enumerate(keys), bs)
+        k => variable(idx = i, bound = b) for ((i, k), b) in Base.zip(enumerate(keys), bs)
     )
 end
 
+"""
+$(TYPEDSIGNATURES)
+
+Allocate a variable for each item in a constraint tree (or any other kind of
+tree) and return a [`ConstraintTree`](@ref) with variables bounded by the
+`makebound` function, which converts a given tree element's value into a bound
+for the corresponding variable.
+"""
+function variables_for(makebound, ts::Tree)
+    var_idx = 0
+    map(ts, Constraint) do x
+        var_idx += 1
+        variable(idx = var_idx, bound = makebound(x))
+    end
+end
```

Comment:
where is this used?

Reply:
i.e. when would I want to use it?

Reply:
https://github.com/LCSB-BioCore/COBREXA.jl/pull/814/files#diff-4073272f80aada4226d99a069713c2e52b6abeb93d66e1124c7f87e3cc9f4abbR48-R54